extension/src/service/web-component-service.js

```diff
@@ -206,18 +207,33 @@ async function fetchAsync(
   const removeSensitiveData =
     requestObj.removeSensitiveData ?? moduleConfig.removeSensitiveData
   delete requestObj.removeSensitiveData
+  let response
+  let responseBody
+  let responseBodyInText
   const request = buildRequest(
     adyenMerchantAccount,
     adyenApiKey,
     requestObj,
     headers
   )
-  const response = await fetch(url, request)
-  const responseBody = await response.json()
-  // strip away sensitive data from the adyen response.
-  request.headers['X-Api-Key'] = '***'
-  if (removeSensitiveData) {
-    delete responseBody.additionalData
+
+  try {
+    response = await fetch(url, request)
+    responseBodyInText = await response.text()
+
+    responseBody = JSON.parse(responseBodyInText)
+  } catch (err) {
+    if (response)
+      // Handle non-JSON format response
+      responseBody = responseBodyInText
```

Comment:
If we do not throw an error here it will be processed as if the Adyen response was successful? Can the caller of this function work with the `responseBodyInText` which it assumed previously to be a valid JSON?

Reply:
Even error is thrown here, the error stack will be serialized as String and return as response in following line
https://github.com/commercetools/commercetools-adyen-integration/blob/e6bf765031d337e87f02bfb3a6ce824dd552ca78/extension/src/service/web-component-service.js#L193

The error stack then is saved into response field in `InterfaceInteraction`. It seems that no difference from without throwing error.

Reply:
I changed to throwing error, as the error stack contains line number for investigation. It would be more useful than simply returning the non-JSON response
https://github.com/commercetools/commercetools-adyen-integration/pull/1059/commits/31d259f17b1e65b5179755c8e407ea666ccd74ec